client/ayon_core/hosts/max/plugins/publish/collect_workfile.py

```diff
@@ -23,15 +24,12 @@ def process(self, context):
 
         context.data['currentFile'] = current_file
 
-        filename, ext = os.path.splitext(file)
-
-        task = context.data["task"]
+        ext = os.path.splitext(file)[-1].lstrip(".")
 
         data = {}
 
         # create instance
-        instance = context.create_instance(name=filename)
-        subset = 'workfile' + task.capitalize()
+        subset = instance.data["subset"]
 
         data.update({
```

Comment:
A lot of the logic here of data being adjusted can be removed since it SHOULD be handled by the Creator.

The subset, asset, publish, family and families should be keys that should be inherited from the creator plug-in's instance as opposed to set here - as far as I know.